src/main/java/it/pagopa/ecommerce/payment/instruments/infrastructure/rule/FilterRuleEngine.java

```diff
@@ -0,0 +1,26 @@
+package it.pagopa.ecommerce.payment.instruments.infrastructure.rule;
+
+import it.pagopa.ecommerce.payment.instruments.infrastructure.PspDocument;
+import it.pagopa.ecommerce.payment.instruments.infrastructure.PspRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import reactor.core.publisher.Flux;
+
+import java.util.List;
+
+@AllArgsConstructor
+public class FilterRuleEngine {
```

Comment:
what do you think if `FilterRuleEngine` became a spring component and use it with `@Autowired`?